brew: make a common page for macOS, Linux and Windows (#5693)
tldr-pages/tldr: pages/common/brew.md

# brew

> > Package manager for macOS (or Linux).

Author: What about this?
```suggestion
> Package manager for macOS, Linux or Windows 10 Subsystem for Linux
```

Reviewer: It also works in BSD derivations and many other systems, and WSL is just Linux in an emulator. Stating those 3 doesn't make much sense.

Author: OK, so does it make sense to merge in a common pages (I don't think there are pages for "macOS and Linux"). :innocent:

Author: If so, what about `Package manager for macOS and Linux` ?

Reviewer 2: I think that's good, @nicokosi 👍🏻

Reviewer 3: What about `Package manager for *nix systems.` ?

Reviewer 2: That seems fine to me, except we need to be careful and make sure that we aren't being inaccurate at all (the Homebrew docs say macOS and Linux, they don't indicate anything else although it may work fine on other platforms)

> > Package manager for macOS (or Linux).

Author: ```suggestion
> Package manager for macOS and Linux.
```

> More information: <https://brew.sh>.

- Search for available formulae (i.e. packages) and casks (i.e. native packages):

`brew search {{text}}`

- Install the latest stable version of a formula or cask (use `--devel` for development versions):

> `brew install {{formula}}`

Reviewer 2: It might be a good idea to switch the positions of these two, because `install` seems more essential than `search` 🤔

Author: Done in fb8a8d39.

- List all installed formulae and casks:

`brew list`

- Upgrade an installed formula or cask (if none is given, all installed formulae/casks are upgraded):

`brew upgrade {{formula}}`

> - Fetch the newest version of Homebrew and of all formulae and casks from GitHub:

Reviewer 2: This confused me for just a second, because it can sound like it downloads them from the tools' respective GitHub repos rather than the main Homebrew repository.  Just a thought.
```suggestion
- Fetch the newest version of Homebrew and of all formulae and casks from the Homebrew source repository:
```

Author: Or this ?
```suggestion
- Fetch the newest version of Homebrew and of all formulae and casks:
```

Reviewer 2: That's fine, but it might be nice to know where they're coming from.  Is there anything about my suggestion that you don't like? (💯 fine if there is)

Author: I find your suggestion OK. :ok_hand:

`brew update`

> - Display information about a formula/cask (version, installation path, dependencies, etc.):

Author: Use the same expression everywhere: `formula/cask` or `formula or cask`?

Reviewer 2: Not really sure.  I think both are fine, maybe @sbrl has a preference?

Reviewer 4: I'd say if they are synonyms for each other, we should go with `formula/cask` and if they are two different things `formula or cask`.

Reviewer 2: The latter is the case, so use `formula or cask`.

Author: Seems OK in ab99cf5b4f789d945082926dfc327e1f63bfab2c.

> > More information: <https://brew.sh>.
> 
> - Search for available formulae (i.e. packages) and casks (i.e. native packages):
> 
> `brew search {{text}}`
> 
> - Install the latest stable version of a formula or cask (use `--devel` for development versions):
> 
> `brew install {{formula}}`
> 
> - List all installed formulae and casks:
> 
> `brew list`
> 
> - Upgrade an installed formula or cask (if none is given, all installed formulae/casks are upgraded):
> 
> `brew upgrade {{formula}}`
> 
> - Fetch the newest version of Homebrew and of all formulae/casks from GitHub:
> 
> `brew update`
> 
> - Display information about a formula/cask (version, installation path, dependencies, etc.):

Author: ```suggestion
> More information: <https://brew.sh>.

- Search for available formulae (i.e. packages) and casks (i.e. native packages):

`brew search {{text}}`

- Install the latest stable version of a formula or cask (use `--devel` for development versions):

`brew install {{formula}}`

- List all installed formulae and casks:

`brew list`

- Upgrade an installed formula or cask (if none is given, all installed formulae/casks are upgraded):

`brew upgrade {{formula}}`

- Fetch the newest version of Homebrew and of all formulae and casks from GitHub:

`brew update`

- Display information about a formula or a cask (version, installation path, dependencies, etc.):
```
